SpatialGDK/Source/SpatialGDKEditor/Private/SpatialTestSettings.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialTestSettings.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialTestSettings);
+
+FSpatialTestSettings::FSpatialTestSettings()
+	: OriginalSpatialGDKSettings(nullptr)
+	, OriginalLevelEditorPlaySettings(nullptr)
+	, OriginalSpatialGDKEditorSettings(nullptr)
+	, OriginalGeneralProjectSettings(nullptr)
+	, OriginalEditorPerformanceSettings(nullptr)
+{
+}
+
+void FSpatialTestSettings::Override(const FString& MapName)
+{
+	// Back up the existing settings so they can be reverted later
+	Duplicate<USpatialGDKSettings>(OriginalSpatialGDKSettings);
+	Duplicate<ULevelEditorPlaySettings>(OriginalLevelEditorPlaySettings);
+	Duplicate<USpatialGDKEditorSettings>(OriginalSpatialGDKEditorSettings);
+	Duplicate<UGeneralProjectSettings>(OriginalGeneralProjectSettings);
+	Duplicate<UEditorPerformanceSettings>(OriginalEditorPerformanceSettings);
+	// First override the settings from the base config file, if it exists
+	if (FPaths::FileExists(BaseOverridesFilename))
+	{
+		Load(BaseOverridesFilename);
+	}
+	else
+	{
+		UE_LOG(LogSpatialTestSettings, Log, TEXT("Base config file not found %s."), *BaseOverridesFilename);
+	}
+	// Then override the settings from the map specific config file, if it exists
+	FString MapOverridesFilename = OverrideSettingsBaseFilename + FPackageName::GetShortName(MapName) + (OverrideSettingsFileExtension);
+	if (FPaths::FileExists(MapOverridesFilename))
+	{
+		Load(MapOverridesFilename);
+	}
+	else
+	{
+		UE_LOG(LogSpatialTestSettings, Log, TEXT("Map specific config file not found %s."), *MapOverridesFilename);
+	}
+	// Then override the settings from the generated map specific config file, if it exists
+	FString GeneratedMapOverridesFilename =
+		GeneratedOverrideSettingsBaseFilename + FPackageName::GetShortName(MapName) + (OverrideSettingsFileExtension);
+	if (FPaths::FileExists(GeneratedMapOverridesFilename))
```

Comment:
This section confuses me a little - why do we have three ini files that can be overridden for a single test?

Reply:
There is a base config that can override the settings for all tests.
Then there is the map level config that overrides the settings for the specific map.
Finally, we added the generated map level config for easier integration with the generated test map configs as they are placed in the intermediate directory.